src/wallet/message_store.cpp

```diff
@@ -182,8 +181,8 @@ bool message_store::signer_labels_complete() const
 void message_store::get_signer_config(std::string &signer_config)
 {
   std::stringstream oss;
-  boost::archive::portable_binary_oarchive ar(oss);
-  ar << m_signers;
+  binary_archive<true> ar(oss);
```

Comment:
`boost::serialization` has an implicit string-header in every archive as a quick sanity check. I recommend the same thing here - it will quickly/correctly reject `boost::serialization` archives, identify incorrectly encoded `binary_archive`'s (this detection cannot be done currently), and will make future format changes easier in the future.

The primary problem is that this format is binary and compact. A future format could overlap significantly with the encoding scheme, resulting in undetected decoding bugs.

The quickest method is likely inheriting from the existing type but with a read/write call in the constructor.